pipenv/core.py

```diff
@@ -1258,19 +1258,33 @@ def do_init(
                     err=True,
                 )
             else:
-                click.echo(
-                    crayons.red(
-                        u"Pipfile.lock ({0}) out of date, updating to ({1})…".format(
-                            old_hash[-6:], new_hash[-6:]
+                if not old_hash:
+                    # Lockfile corrupted, remove it and replaced
+                    click.echo(
+                        crayons.red(
+                            u"Pipfile.lock is corrupted, replaced with ({0})…".format(
+                                new_hash[-6:]
+                            ),
+                            bold=True,
                         ),
-                        bold=True,
-                    ),
-                    err=True,
-                )
+                        err=True
+                    )
+                    os.remove(project.lockfile_location)
```

Comment:
Wait why are we deleting this?

Reply:
We intentionally _don't_ explicitly delete lockfiles unless we are finished re-locking and have a new one ready to replace them with.  There is never a scenario where it is OK to just delete a lockfile -- it should all be handled in the project internals. Other than that one line I'm good with the changes, but @uranusjr wouldn't this have been addressed by introducing the lockfile decoder?

Reply:
In this particular case I feel it’s fine to delete the lock file because it is corrupted, but either way would do.

Reply:
'Corrupted' as in doesn't have a hash in it? Doesn't have content? How does this state occur? Either way I really can't see any reason why we would _ever_ forcibly delete the only file in pipenv we made special functionality to ensure never gets removed without being replaced.  

This is super inconsistent.  If there is a weird problem in a lockfile users should have a chance to modify or mess with or backup their lockfile unless they successfully ran a new lock, which should be _the only thing in all of pipenv_ that removes lockfiles

Reply:
No. Do not manipulate the lockfile. We already have tools that do this correctly when you lock. I see no reason to complicate the process with extra manipulation when we already manage all of this. This is a landmine that secretly deletes something that would already be deleted during locking if it succeeded. If it didn’t succeed it would be a lot more helpful if we don’t randomly manipulate it in just this one spot.

Reply:
No. Do not manipulate the lockfile. We already have tools that do this correctly when you lock. I see no reason to complicate the process with extra manipulation when we already manage all of this. This is a landmine that secretly deletes something that would already be deleted during locking if it succeeded. If it didn’t succeed it would be a lot more helpful if we don’t randomly manipulate it in just this one spot.

Reply:
For my particular context “corrupted” means the file cannot be JSON decoded. But the `os.remove` line is not strictly necessary either way. I’m +0 on removing it.